### PR TITLE
Add types for fast-decode-uri-component

### PR DESCRIPTION
**Pull request by huseeiin**
Add types for fast-decode-uri-component

Please fill in this template.

- [ ] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [ ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:

- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:

- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

If removing a declaration:

- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/fast-decode-uri-component/package.json

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against c1fcf6db3d0cffa9f5834259658c6d8d7a52587c
</p>

File: types/fast-decode-uri-component/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+        "noImplicitAny": true,
+        "noImplicitThis": true,
+        "strictNullChecks": true,
+        "strictFunctionTypes": true,
+        "types": [],
+        "noEmit": true,
+        "forceConsistentCasingInFileNames": true
+    },
+    "files": [
+        "index.d.ts"
```
Comment (sheetalkamat) [inline types/fast-decode-uri-component/tsconfig.json:16]: pls add a test

**Review by sheetalkamat — CHANGES_REQUESTED** [review]

File: types/fast-decode-uri-component/tsconfig.json (hunk 1 above)
Comment (sheetalkamat) [reply types/fast-decode-uri-component/tsconfig.json:16]: You still need to add test file name here . Please follow https://github.com/DefinitelyTyped/DefinitelyTyped?tab=readme-ov-file#create-a-new-package when adding new package

**Review by sheetalkamat — CHANGES_REQUESTED** [review]

**Review by sheetalkamat — APPROVED** [review]

**Comment by huseeiin** [comment]
Ready to merge

**Pull request merged**